backend/src/main/java/event/router/SimplePlaylistLoad.java

```diff
@@ -0,0 +1,57 @@
+package event.router;
+
+import gpio.GpioService;
+import gpio.PlaylistLoadAction;
+import mpd.MpdService;
+import rfid.RfidService;
+import rfid.RfidTagUid;
+
+import javax.inject.Inject;
+
+import java.util.Map;
+
+import static hal.JbertPlaylistSelectButton.BLACK;
+import static hal.JbertPlaylistSelectButton.BLUE;
+import static hal.JbertPlaylistSelectButton.GREEN;
+import static hal.JbertPlaylistSelectButton.RED;
+import static hal.JbertPlaylistSelectButton.YELLOW;
+
+
+/**
+ * Event router implementation loading playlists by GPI and RFID event
+ */
+public class SimplePlaylistLoad implements EventRouter {
+    private final Map<RfidTagUid, String> rfidTagMapping;
+    private final EventRouterUtil eventRouterUtil;
+
+    // Event sources
+    private final RfidService rfidService;
+    private final GpioService gpioService;
+
+    // Event sinks
+    private final MpdService mpdService;
+
+    public SimplePlaylistLoad(Map<RfidTagUid, String> rfidTagMapping, RfidService rfidService, GpioService gpioService, MpdService mpdService) {
+        this.rfidTagMapping = rfidTagMapping;
+        this.eventRouterUtil = new EventRouterUtil(rfidService, mpdService);
+        this.rfidService = rfidService;
+        this.gpioService = gpioService;
+        this.mpdService = mpdService;
+    }
+
+    @Override
+    public void configure() {
+        eventRouterUtil.registerPlaylistChangeActions(rfidTagMapping, true);
+        eventRouterUtil.configureControlGpiListener(gpioService, mpdService);
+
+        configureGpioEventHandling();
+    }
+
+    private void configureGpioEventHandling() {
+        gpioService.registerGpiListener(RED.getGpioName(), new PlaylistLoadAction(mpdService, "red"));
+        gpioService.registerGpiListener(BLACK.getGpioName(), new PlaylistLoadAction(mpdService, "black"));
+        gpioService.registerGpiListener(GREEN.getGpioName(), new PlaylistLoadAction(mpdService, "green"));
+        gpioService.registerGpiListener(BLUE.getGpioName(), new PlaylistLoadAction(mpdService, "blue"));
+        gpioService.registerGpiListener(YELLOW.getGpioName(), new PlaylistLoadAction(mpdService, "yellow"));
```

Comment:
This looks like being easily replaceable by a iteration over the enum values:

```
EnumSet.allOf(JbertPlaylistSelectButton.class).forEach(...)
```

Reply:
nice! done!